src/Mod/Path/PathScripts/PathProfile.py

```diff
@@ -322,12 +324,16 @@ def areaOpShapes(self, obj):
 
         # Pre-process Base Geometry to process edges
         if obj.Base and len(obj.Base) > 0:  # The user has selected subobjects from the base.  Process each.
-            shapes.extend(self._processEdges(obj))
+            edge_shapes, otherBase = self._processEdges(obj)
+            shapes.extend(edge_shapes)
 
-        if obj.Base and len(obj.Base) > 0:  # The user has selected subobjects from the base.  Process each.
+        if otherBase is False:
```

Comment:
`otherBase` gets initialized with `None`, and now it's compared to `False` ?

it would be better if `otherBase` gets intialized to `[]` and `_processEdges` also returns an empty list - variables that switch types are a recipe for pain.

Reply:
Yes, I will revise and use the same data type throughout uses of `newBase` and related `otherBase`.